beacon_node/beacon_chain/src/validator_monitor.rs

```diff
@@ -394,9 +413,32 @@ impl<T: EthSpec> ValidatorMonitor<T> {
         });
     }
 
+    /// Add an unaggregated attestation
+    pub fn set_unaggregated_attestation(&mut self, attestation: Attestation<T>) {
+        let unaggregated_attestations = &mut self.unaggregated_attestations;
+
+        // Pruning, this removes the oldest key/pair of the hashmap if it's greater than MAX_UNAGGREGATED_ATTESTATION_HASHMAP_LENGTH
+        if unaggregated_attestations.len() >= MAX_UNAGGREGATED_ATTESTATION_HASHMAP_LENGTH {
+            if let Some(oldest_slot) = unaggregated_attestations.keys().min() {
+                unaggregated_attestations.remove(&oldest_slot.clone());
+            }
```

Comment:
```suggestion
            if let Some(oldest_slot) = unaggregated_attestations.keys().min().copied() {
                unaggregated_attestations.remove(&oldest_slot);
            }
```

This is slightly more idiomatic since it uses `Copy` rather than `Clone`. Just a nitpick.